src/ua/artcode/udiary/model/User.java

```diff
@@ -20,11 +21,13 @@
     // Constructors:
 
     public User() {
```

Comment:
User must have id field, consequently, we can use the only id in overriding equals and hashcode

Reply:
I agree with you about hashCode(), but I think in some cases (using collections, validators, etc.) it is better to use equals by all fields except id. We assign id only when we store user in DB - it means that actually equal objects will be not equal by equals, because id is unique or is not assigned yet for one of User objects we compare.